src/WebSocket.cpp

```diff
@@ -792,10 +813,9 @@ namespace WebSocket
                         }
                     } // while
                     if (headersFinished) {
-                        if (headers.isEmpty()
-                                || headers.value(QStringLiteral("upgrade")).toLower() != QStringLiteral("websocket")
-                                || headers.value(QStringLiteral("connection")).toLower() != QStringLiteral("upgrade"))
-                            Fail("Bad header fields", 426, "Upgrade Required");
+                        QString message;
+                        if (!checkHeaders(&message))
```

Comment:
Tiny nit: I’m a huge fan of the C++17 scoped-if delcarator feature.. mind putting the declaration of `message` in the `if`?

```c++
If (QString message; !checkHeaders(&message)) ....
```
If you don’t change it.. I will after merge. :)

Reply:
TIL this is a thing. Done.